src/main/java/ru/spbstu/exception/GlobalExceptionHandler.java

```diff
@@ -25,4 +26,10 @@ public GlobalApiErrorResponse dataIntegrityViolationException(DataIntegrityViola
   public GlobalApiErrorResponse fileProcessingException(FileProcessingException fe) {
     return new GlobalApiErrorResponse(HttpStatus.UNPROCESSABLE_ENTITY, fe.getMessage());
   }
+
+  @ExceptionHandler(UsernameNotFoundException.class)
+  @ResponseStatus(HttpStatus.NOT_FOUND)
+  public GlobalApiErrorResponse userNameNotFoundException(UsernameNotFoundException un) {
+    return new GlobalApiErrorResponse(HttpStatus.NOT_FOUND, un.getMessage());
```

Comment:
Imho, HttpStatus.UNAUTHORIZED would be more suitable here.

Reply:
In my opinion `UNAUTHORIZED` is more suitable for cases when user forgot to authenticate isn't it?